junit-platform-launcher/src/main/java/org/junit/platform/launcher/core/EngineDiscoveryOrchestrator.java

```diff
@@ -82,7 +86,10 @@ public LauncherDiscoveryResult discover(LauncherDiscoveryRequest request, String
 			testEngineDescriptors.put(testEngine, rootDescriptor);
 		}
 
-		applyPostDiscoveryFilters(testEngineDescriptors, request.getPostDiscoveryFilters());
+		final List<PostDiscoveryFilter> filters = new LinkedList<>(postDiscoveryFilters);
```

Comment:
nit: we don't use `final` for local variables.
```suggestion
		List<PostDiscoveryFilter> filters = new LinkedList<>(postDiscoveryFilters);
```